shared/tests/utility/math/TestCoordinates.cpp

```diff
@@ -0,0 +1,1029 @@
+    Eigen::Matrix<double, 3, 1>(-7.641797112946607e+18, 1.141808506244429e+19, -1.223888354349969e+19),
+    Eigen::Matrix<double, 3, 1>(1.239274786533388e+18, 7.231310491504572e+17, 1.219815117843094e+17),
+    Eigen::Matrix<double, 3, 1>(-4.583839400878947e+18, -3.684839294682394e+18, -5.670914892617796e+18),
+    Eigen::Matrix<double, 3, 1>(-1.622791603116365e+18, 1.083462077044538e+18, 2.712145653561990e+17),
+    Eigen::Matrix<double, 3, 1>(-5.272487849080346e+18, 4.973099800345554e+18, 1.614803858854391e+19),
+    Eigen::Matrix<double, 3, 1>(-5.983154538657204e+15, 1.319502776222251e+16, 8.375615263442275e+16),
+    Eigen::Matrix<double, 3, 1>(-8.999363070143180e+17, 1.333121729459132e+19, 5.094974002652679e+18),
+    Eigen::Matrix<double, 3, 1>(-9.592444819019354e+18, -6.463145612895186e+18, 9.706840432426117e+18),
+    Eigen::Matrix<double, 3, 1>(-1.494601059566613e+19, -2.129089221787754e+18, -5.299410001113628e+18),
+    Eigen::Matrix<double, 3, 1>(-7.733947778490496e+17, 3.924619250990513e+17, 1.296701258942616e+18),
+    Eigen::Matrix<double, 3, 1>(4.503738767552534e+17, -1.082993713041953e+18, -7.286198459323493e+18),
+    Eigen::Matrix<double, 3, 1>(-4.793668514184680e+17, -7.711036082725997e+17, -4.703159219829807e+18),
+    Eigen::Matrix<double, 3, 1>(-4.906112178815113e+18, -1.021906694429197e+19, 9.515289489918910e+18),
+    Eigen::Matrix<double, 3, 1>(-6.695147848240914e+17, 3.523887917749655e+17, -7.923961893669941e+18),
+    Eigen::Matrix<double, 3, 1>(8.649996401425574e+18, 5.817999453015102e+18, -1.317453773838158e+19),
+    Eigen::Matrix<double, 3, 1>(2.409796589198986e+18, 1.716766783429139e+18, -1.592228692744638e+18),
+    Eigen::Matrix<double, 3, 1>(4.781741493243726e+18, -8.834397216735164e+17, -2.671751297570155e+17),
+    Eigen::Matrix<double, 3, 1>(1.126088014672571e+17, 3.187572358457569e+17, -2.668672531883236e+18)
+
+};
+
+/** Test cartesianToSpherical conversion */
+TEST_CASE("Test coordinate conversion - Cartesian to spherical.", "[utility][math][coordinates]") {
+    // Loops through test values and compare to results array
+    for (size_t i = 0; i < cart_coords.size(); i++) {
+        Eigen::Vector3d cart_input   = cart_coords[i];
+        Eigen::Vector3d cart_compare = cartToSpher_results[i];
+        Eigen::Vector3d spher_result = utility::math::coordinates::cartesianToSpherical(cart_input);
+        Approx epsilon_x             = Approx(cart_compare.x()).epsilon(ERROR_THRESHOLD);
+        Approx epsilon_y             = Approx(cart_compare.y()).epsilon(ERROR_THRESHOLD);
+        Approx epsilon_z             = Approx(cart_compare.z()).epsilon(ERROR_THRESHOLD);
+        //***Output table***
+        INFO(fmt::format("{:*^50}", "Calculating Spherical coordinates for the origin"));
+        INFO(fmt::format("{:-^50}", ""));
+        INFO(fmt::format("Input:\n x = {}\n y = {}\n z = {}", cart_input.x(), cart_input.y(), cart_input.z()));
```

Comment:
99% sure that if you use [fmt ostream](https://fmt.dev/latest/api.html#ostream-api) you cant print `Eigen` vectors and matrices directly (might need to add a transpose as well)
```suggestion
        INFO(fmt::format("Input: {}", cart_input.tranpose()));
```